couchdb/server/state.py

```diff
@@ -2,11 +2,37 @@
 #
 import logging
 
-__all__ = ('add_fun', 'add_lib')
+__all__ = ('add_fun', 'add_lib', 'reset')
 
 log = logging.getLogger(__name__)
 
 
+def reset(server, config=None):
+    """Resets query server state.
+
+    :command: reset
+
+    :param server: Query server instance.
+    :type server: :class:`~couchdb.server.BaseQueryServer`
+
+    :param config: Optional dict argument to set up query config.
+    :type config: dict
+
+    :return: True
+    :rtype: bool
+    """
+    log.debug('Reset server state')
+    del server.state['functions'][:]
+    del server.state['functions_src'][:]
+    server.state['query_config'].clear()
+    if config is not None:
+        log.debug('Set new query config:\n%s', config)
+        server.state['query_config'].update(config)
+    if server.version >= (1, 1, 0):
+        server.state['view_lib'] = ''
+    return True
```

Comment:
We init `server.state` in the constructor of `BaseQueryServer`.
```python
        self._state = {
            'view_lib': None,
            'line_length': 0,
            'query_config': {},
            'functions': [],
            'functions_src': [],
            'row_line': {}
        }
```

But in this `reset` function, we do not reset the value of `line_length` and `row_line`.

About `line_length`: I cannot find any function need it. Is this variable an unimplemented feature? 

Following `grep` shows that only docstring mensioned it.
It's the partner of `reduce_limilt`, but, the `reduce_limit` cannot be configured from commandline optoin.
Is `reduce_limit` an unimplemented feature, also?
```shell
└─[iblis@imfsa]% grep -1 line_length *
__init__.py-            'view_lib': None,
__init__.py:            'line_length': 0,
__init__.py-            'query_config': {},
Binary file __init__.pyc matches
--
views.py-          If any Python exception occurs or reduce output is twice longer
views.py:          as state.line_length and reduce_limit is enabled in state.query_config
views.py-    """
--
views.py-          If any Python exception occurs or reduce output is twice longer
views.py:          as state.line_length and reduce_limit is enabled in state.query_config
views.py-    """
Binary file views.pyc matches
```